homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -71,6 +75,13 @@
         vol.All(vol.Coerce(int), vol.Clamp(min=-179, max=179)),
     vol.Optional(ATTR_RC_DURATION): cv.positive_int,
 })
+SERVICE_SCHEMA_CLEAN_ZONE = VACUUM_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_ZONE_ARRAY):
+        vol.All(list, [vol.ExactSequence([int, int, int, int])]),
```

Comment:
Make sure to coerce the sequence values to integers.
```suggestion
        vol.All(list, [vol.ExactSequence([vol.Coerce(int), vol.Coerce(int), vol.Coerce(int), vol.Coerce(int)])]),
```

Reply:
fixed